src/main/java/com/google/firebase/remoteconfig/Condition.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Strings;
+import com.google.firebase.internal.NonNull;
+import com.google.firebase.internal.Nullable;
+import com.google.firebase.remoteconfig.internal.TemplateResponse.ConditionResponse;
+
+/**
+ * Represents a Remote Config condition that can be included in a {@link Template}.
+ * A condition targets a specific group of users. A list of these conditions make up
+ * part of a Remote Config template.
+ */
+public final class Condition {
+
+  private String name;
+  private String expression;
+  private TagColor tagColor;
+
+  /**
+   * Creates a new {@link Condition}.
+   *
+   * @param name A non-null, non-empty, and unique name of this condition.
+   * @param expression A non-null and non-empty expression of this condition.
+   */
+  public Condition(@NonNull String name, @NonNull String expression) {
```

Comment:
I think we need a constructor that accepts the color as well.

Reply:
Added a constructor with color. Should we amend the API proposal as well?

Reply:
Feel free to include a note. Looks like an oversight if it's not already there.